apps/prairielearn/src/pages/instructorStudents/instructorStudents.tsx

```diff
@@ -20,6 +21,11 @@ const sql = loadSqlEquiv(import.meta.url);
 
 router.get(
   '/',
+  createAuthzMiddleware({
+    oneOfPermissions: ['has_course_instance_permission_view'],
+    errorMessage: 'Access denied (must be a student data viewer)',
+    unauthorizedUsers: 'passthrough',
+  }),
```

Comment:
Here and elsewhere, I can't tell why we have both this middleware (with `unauthorizedUsers: 'passthrough'`) and the auth checks inside the handler. Are you thinking at all about making `InsufficientCoursePermissionsCardPage` and such part of `createAuthzMiddleware?

Reply:
I was thinking about that, but it felt out of scope for what this PR was aiming to accomplish. Those pages also require information about e.g. the course id. I would certainly prefer to do that in the future, and it would mean that a lot more code can be reused.

Reply:
I immediately needed this for #12399 .